src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -630,6 +631,214 @@ def assign_second_guardian_to_newborns(
+            race_college_sims_with_single_female_reference_person_idx = (
+                race_college_sims_idx.intersection(
+                    college_sims_with_single_female_reference_guardian_idx
+                )
+            )
+            race_college_sims_with_single_male_reference_person_idx = (
+                race_college_sims_idx.intersection(
+                    college_sims_with_single_male_reference_guardian_idx
+                )
+            )
+            race_college_sim_with_partnered_reference_person_idx = (
+                race_college_sims_idx.intersection(
+                    college_sims_with_partnered_reference_guardian_idx
+                )
+            )
+
+            # Get ids (index) for each possible reference person guardian type
+            race_female_single_reference_person_ids = (
+                households_with_single_female_reference_person.loc[
+                    households_with_single_female_reference_person["race_ethnicity"] == r,
+                    "reference_person_id",
+                ]
+            )
+            race_male_single_reference_person_ids = (
+                households_with_single_male_reference_person.loc[
+                    households_with_single_male_reference_person["race_ethnicity"] == r,
+                    "reference_person_id",
+                ]
+            )
+            race_partnered_households = households_with_partners.loc[
+                households_with_partners["race_ethnicity"] == r
+            ]
+
+            # Choose and assign guardian_1
+            single_female_guardian_ids = self.randomness.choice(
+                race_college_sims_with_single_female_reference_person_idx,
+                choices=race_female_single_reference_person_ids,
+                additional_key=f"{r}_single_female_guardian_ids",
+            )
+            single_male_guardian_ids = self.randomness.choice(
+                race_college_sims_with_single_male_reference_person_idx,
+                choices=race_male_single_reference_person_ids,
+                additional_key=f"{r}_single_male_guardian_ids",
+            )
+            partnered_guardian_ids = self.randomness.choice(
+                race_college_sim_with_partnered_reference_person_idx,
+                choices=race_partnered_households["reference_person_id"],
+                additional_key=f"{r}_partnered_guardian_ids",
+            )
+            pop.loc[
+                single_female_guardian_ids.index, "guardian_1"
+            ] = single_female_guardian_ids
+            # Should I use the original index of race_college_sims_with_single_female_reference_person_idx for lookup
+            pop.loc[single_male_guardian_ids.index, "guardian_1"] = single_male_guardian_ids
+            pop.loc[partnered_guardian_ids.index, "guardian_1"] = partnered_guardian_ids
+            # Assign guardian_2 for sims who's reference person has a partner
+            if len(race_partnered_households) > 0:
```

Comment:
This is a check to make sure we do not have an empty dataframe which would break the below call that applies np.random.choice